launch/launch/__init__.py

```diff
@@ -75,8 +78,11 @@ def __init__(self, cmd, name, output_handler, exit_handler, env=None):
 
     def send_signal(self, signal):
         if self.transport:
+            self.task_state.signals_received.append(signal)
             self.transport.send_signal(signal)
 
     def terminate(self):
-        if self.transport:
+        if os.name != 'nt':
+            self.send_signal(signal.SIGINT)
```

Comment:
Currently `terminate` is being called on the task descriptor when the expected output is matched. If we assume that the process is not handling SIGINT perfectly then this change renders that `terminate` call useless. Instead it relies on the launcher to cleanup all process afterwards.

From my point of view the current patch is not consistent. Either the `terminate` method should be removed from the task descriptor (since its not doing anymore what it is supposed to) or it needs to keep actually terminating the subprocess. If we keep the responsibility to terminate processes to the launch only then the task descriptor should not have a `terminate` method.

Also the launcher enforces `SIGINT` and `SIGTERM` on all processes after exiting the main loop. So a process might hang around for a long time if the lanuch file is supposed to be running longer than one individual process.